MediaBrowser.Providers/Lyric/TxtLyricProvider.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections.Generic;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Lyrics;
+using MediaBrowser.Controller.Resolvers;
+
+namespace MediaBrowser.Providers.Lyric;
+
+/// <summary>
+/// TXT Lyric Provider.
+/// </summary>
+public class TxtLyricProvider : ILyricProvider
+{
+    /// <inheritdoc />
+    public string Name => "TxtLyricProvider";
+
+    /// <summary>
+    /// Gets the priority.
+    /// </summary>
+    /// <value>The priority.</value>
+    public ResolverPriority Priority => ResolverPriority.Second;
+
+    /// <inheritdoc />
+    public IReadOnlyCollection<string> SupportedMediaTypes { get; } = new[] { "lrc", "txt" };
+
+    /// <summary>
+    /// Opens lyric file for the requested item, and processes it for API return.
+    /// </summary>
+    /// <param name="item">The item to to process.</param>
+    /// <returns>If provider can determine lyrics, returns a <see cref="LyricResponse"/>; otherwise, null.</returns>
+    public LyricResponse? GetLyrics(BaseItem item)
+    {
+        string? lyricFilePath = LyricInfo.GetLyricFilePath(this, item.Path);
+
+        if (string.IsNullOrEmpty(lyricFilePath))
+        {
+            return null;
+        }
+
+        string[] lyricTextLines = System.IO.File.ReadAllLines(lyricFilePath);
+
+        List<LyricLine> lyricList = new List<LyricLine>();
+
+        if (lyricTextLines.Length == 0)
+        {
+            return null;
+        }
+
+        foreach (string lyricTextLine in lyricTextLines)
+        {
+            lyricList.Add(new LyricLine(lyricTextLine));
```

Comment:
This is a great example of `IReadOnlyList` being the superior choice. This won't be necessary as you can just assign it to the `Lyrics` property directly. (But pretending that it had to be a list, a simple `.ToList` would be better than adding to a new list).

Reply:
"This won't be necessary as you can just assign it to the Lyrics property directly."

This is my C# ignorance talking, but how can you directly assign it to `Lyrics` property when `lyricTextLines` is `string[]` but the `Lyrics` property  is wanting `IReadOnlyList<LyricLine>`?

Reply:
An array implements `IReadOnlyList`

Reply:
I mean the data itself. The loop is to convert from string to `Lyricline`. How can that be avoided?

Reply:
Right, I was sleepy. It can't be avoided, but you can avoid the use of .Add creating a new array under the hood by either specifying the List size or by creating an array of size `lyricTextLines,Length`